### PR TITLE
Implement RSS/Atom feeds

### LINKED ISSUE #159: RSS feed for news articles
RSS / Atom feeds should be fairly easy to generate statically, given the metadata provided in news article pages.

### PR DESCRIPTION
**Pull request by JMAlego**
Implementation for RSS/Atom feeds for the website.

- [x] RSS feed
- [x] Atom feed
- [x] Work out correct way to get absolute URL from flask, context.yaml possible?
- [x] Validate generated XML as RSS/Atom

Fixes #159

### CODE REVIEW

**Comment by JMAlego** [comment]
@LukeMoll do you know the correct way to generate an absolute URL from flask? During development or when running dynamically you can use `url_for` with the `_external=True` parameter (which feels wrong as it is). When freezing we probably need to store the absolute URL in somewhere like `context.yaml` right?

**Comment by LukeMoll** [comment]
What feels wrong about `_external`? This seems like the most useful way to do it in Flask.

Given the GitHub repository is hard-coded in filters.py, I'd be comfortable hard-coding the website URL.

As per [https://flask.palletsprojects.com/en/2.0.x/api/#flask.url_for](https://flask.palletsprojects.com/en/2.0.x/api/#flask.url_for), we can set the URL root with `SERVER_NAME`. The tricky bit would be handling the difference between development and production environments; the former probably wants it set to `localhost:8000` (icr the port number), but the latter would use the actual webroot. We could set it to always use the actual webroot, but this could hinder development with absolute URLs.

The note in the docs that 
> Configuration values APPLICATION_ROOT and SERVER_NAME are only used when generating URLs outside of a request context.

is concerning, as AIUI we'd always be generating in a request context.

**Comment by LukeMoll** [comment]
Thanks for working on this, I'll try to give it a good look over on Monday. One quick comment based on the discussion here is that the cannonical URL for the website is *www.* HackSoc.org, so the base URL should probably reflect that. nginx is configured to redirect anyway so it shouldn't matter too much.

**Review by LukeMoll — CHANGES_REQUESTED** [review]
Other than the minor documentation suggestion, looks good to me, thank you! Have tested RSS and Atom in Akregator and behave as expected.

File: hacksoc_org/routes.py [hunk 1]
```diff
@@ -140,3 +140,29 @@ def render_news(article: str):
...
         f"content/news/{article}.html.jinja2", date=date.fromisoformat(article[:10])
     )
+
+
+@blueprint.route("/rss.xml")
+def render_rss_feed():
+    """Render the rss feed using template.
+
+    Returns:
+        str: Full HTML page
```
[the remark this answers is not in the document]
Comment (JMAlego) [reply hacksoc_org/routes.py:150]: Ah good spot

**Review by LukeMoll — APPROVED** [review]

**Pull request merged**